esi/src/lib.rs

```diff
@@ -97,96 +104,370 @@ impl Processor {
+            }
+            Element::Try {
+                mut attempt_task,
+                mut except_task,
+            } => {
+                *depth += 1;
+                process_try(
+                    task,
+                    output_writer,
+                    &mut attempt_task,
+                    &mut except_task,
+                    depth,
+                    dispatch_fragment_request,
+                    process_fragment_response,
+                )?;
+                *depth -= 1;
+                if *depth == 0 {
+                    debug!(
+                        "Writing try result: {:?}",
+                        String::from_utf8(task.output.get_mut().as_slice().to_vec())
+                    );
+                    output_handler(output_writer, task.output.get_mut().as_ref())?;
+                    task.output.get_mut().clear();
+                }
+            }
+        }
+    }
+    Ok(FetchState::Succeeded)
+}
+
+fn process_include(
+    task: &mut Task,
+    fragment: Fragment,
+    output_writer: &mut Writer<impl Write>,
+    depth: usize,
+    dispatch_fragment_request: &FragmentRequestDispatcher,
+    process_fragment_response: Option<&FragmentResponseProcessor>,
+) -> Result<FetchState> {
+    // take the fragment and deconstruct it
+    let Fragment {
+        mut request,
+        alt,
+        continue_on_error,
+        pending_request,
+    } = fragment;
+
+    // wait for `<esi:include>` request to complete
+    let resp = match pending_request.wait() {
+        Ok(r) => r,
+        Err(err) => return Err(ExecutionError::RequestError(err)),
+    };
+
+    let processed_resp = if let Some(process_response) = process_fragment_response {
+        process_response(&mut request, resp)?
+    } else {
+        resp
+    };
+
+    // Request has completed, check the status code.
+    if processed_resp.get_status().is_success() {
+        if depth == 0 && task.output.get_mut().is_empty() {
+            debug!("Include is not nested, writing content to the output stream");
+            output_handler(output_writer, &processed_resp.into_body_bytes())?;
+        } else {
+            debug!("Include is nested, writing content to a buffer");
+            task.output
+                .get_mut()
+                .extend_from_slice(&processed_resp.into_body_bytes());
+        }
+
+        Ok(FetchState::Succeeded)
+    } else {
+        // Response status is NOT success, either continue, fallback to an alt, or fail.
+        if let Some(request) = alt {
+            debug!("request poll DONE ERROR, trying alt");
+            if let Some(fragment) =
+                send_fragment_request(request?, None, continue_on_error, dispatch_fragment_request)?
+            {
+                task.queue.push_front(Element::Include(fragment));
+                return Ok(FetchState::Pending);
+            }
+            debug!("guest returned None, continuing");
+            // continue
+            return Ok(FetchState::Succeeded);
```

Comment:
Can this be removed?
```suggestion
            return Ok(FetchState::Succeeded);
```